### PR TITLE
New package: MGIS

### PR DESCRIPTION
**Pull request by thelfer**
The `MFrontGenericInterfaceSupport` project (`MGIS`) aims at proving tools (functions, classes, bindings, etc...) to handle behaviours written using `MFront` generic interface. For information about `MFront`, see <http://tfel.sourceforge.net>. `MFront` is available in `spack` through the `TFEL` package.

​Those tools are meant to be used by (`FEM`, `FFT`, etc.) solver developers. Permissive licences have been chosen to allow integration in open-source and proprietary codes.

### CODE REVIEW

File: var/spack/repos/builtin/packages/mgis/package.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    def cmake_args(self):
+
+        args = []
+
+        for i in ['c', 'fortran', 'python']:
+            if '+' + i + '_bindings' in self.spec:
+                args.append("-Denable-{0}-bindings=ON".format(i))
+            else:
+                args.append("-Denable-{0}-bindings=ON".format(i))
```
Comment (healther) [inline var/spack/repos/builtin/packages/mgis/package.py]: I'm pretty sure you mean `=OFF` here

Comment (thelfer) [reply var/spack/repos/builtin/packages/mgis/package.py]: Well, I am pretty sure that you're right. Thank you very much. This is fixed now.

File: var/spack/repos/builtin/packages/mgis/package.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    version('1.0', sha256='279c98da00fa6855edf29c2b8f8bad6e7732298dc62ef67d028d6bbeaac043b3')
+
+    # variants
+
+    variant('c_bindings', default=True,
+            description='Enables c bindings')
+    variant('fortran_bindings', default=True,
+            description='Enables fortran bindings')
+    variant('python_bindings', default=True,
+            description='Enables python bindings')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mgis/package.py]: I would be fine with dropping `bindings` from all of these variant names. It's usually implied. For example, most packages that add optional Python bindings just have a `+python` variant.

File: var/spack/repos/builtin/packages/mgis/package.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    # dependencies
+
+    depends_on('tfel@3.2.0', when="@1.0")
+    depends_on('tfel@rliv-3.2', when="@rliv-3.2")
+    depends_on('tfel@master', when="@master")
+
+    depends_on('python', when='+python_bindings')
+    depends_on('boost+python', when='+python_bindings')
+
+    extends('python', when='+python')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mgis/package.py]: `extends` implies `depends_on`, so you can drop `depends_on` above.

Comment (thelfer) [reply var/spack/repos/builtin/packages/mgis/package.py]: Done. Thanks for the review.

File: var/spack/repos/builtin/packages/mgis/package.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    depends_on('tfel@master', when="@master")
+    depends_on('boost+python', when='+python')
+    extends('python', when='+python')
+
+    def cmake_args(self):
+
+        args = []
+
+        for i in ['c', 'fortran', 'python']:
+            if '+' + i + in self.spec:
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mgis/package.py]: There is an extra `+` here that needs to be removed.

**Pull request merged**